File: ProblemS2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int what_grade_is(int letters, int sentences, int words);
int count_letters(string word);
int count_words(string word);
int count_sentences(string word);

int main(void)
{
    string phrase = get_string("Text: ");
    int letters = count_letters(phrase);
    int sentences = count_sentences(phrase);
    int words = count_words(phrase);
    int grade = what_grade_is(letters, sentences, words);
    if (grade <= 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

int what_grade_is(int letters, int sentences, int words)
{
    float L = (letters / (float)words) * 100;
    float S = (sentences / (float)words) * 100;
    int grade = round(0.0588 * L - 0.296 * S - 15.8);
    return grade;
}

int count_letters(string word)
{
    int count = 0;
    for (int i = 0; i < strlen(word); i++)
    {
        if (isupper(word[i]))
        {
            count += 1;
        }
        if (islower(word[i]))
        {
            count += 1;
        }
    }
    return count;
}

int count_words(string word)
{
    int spaces = 0;
    for (int i = 0; i < strlen(word); i++)
    {
        if (word[i] == ' ')
        {
            spaces += 1;
        }
    }
    return spaces + 1;
}

int count_sentences(string word)
{
    int count = 0;
    for (int i = 0; i < strlen(word); i++)
    {
        if (word[i] == '.')
        {
            count += 1;
        }
        if (word[i] == '?')
        {
            count += 1;
        }
        if (word[i] == '!')
        {
            count += 1;
        }
    }
    return count;
}

File: ProblemS6/sentimental-readability/readability.py
from cs50 import get_string

def whatGrade(numWords, numLetters, numSentences):
    L = numLetters / numWords * 100
    S = numSentences / numWords * 100
    grade = round(0.0588 * L - 0.296 * S - 15.8)
    return grade

def countLetters(text):
    l = 0
    for c in text:
        if c.isalpha():
            l += 1
    return l

def countWords(text):
    w = 0
    for c in text:
        if c == " ":
            w += 1
    w += 1
    return w

def countSentences(text):
    sentence = 0
    for c in text:
        if c == "." or c == "?" or c == "!":
            sentence += 1
    return sentence

def main():
    text = get_string("Text: ")

    numLetters = countLetters(text)
    numWords = countWords(text)
    numSentences = countSentences(text)

    grade = whatGrade(numWords, numLetters, numSentences)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 15:
        print("Grade 16+")
    else:
        print("Grade: {}".format(grade))

main()
